### ISSUE TITLE
validation_fraction=0 is overridden by the 0.1 default

### ISSUE THREAD

**Issue by vijaydevmasters** [f-inverse/jammi-ai#347]
## Summary

Setting `validation_fraction=0` in a `FineTuneConfig` does not disable the validation split. Training still holds out approximately 10% of the data, and the early-stopping monitor remains `ValLoss`, which produces a monitor against an empty or near-empty validation set.

## Observed behaviour

A job submitted with `validation_fraction=0` logs a validation loss each epoch as if a 10% split had been created. The warmup-step calculation (`warmup_steps=19810 of ~198100`) is based on the full dataset, not the 90% training partition, indicating the split is not actually 0.

## Root cause

Two independent problems compound:

**1. Serde default:** `FineTuneConfig.validation_fraction` has a struct-level `Default` of `0.1` (`crates/jammi-wire/src/fine_tune.rs`, line 390). When the Python client or gRPC proto layer omits the field (the typical encoding for "user did not set this"), it deserialises as `0.1`, not `0.0`. A user who explicitly sets `0` in Python sees `0.1` on the Rust side unless the serialised payload includes an explicit `0.0`.

**2. Coupled metric default:** `early_stopping_metric` defaults to `ValLoss`. Specifying `validation_fraction=0.0` without also overriding the metric causes the trainer to monitor a validation loss that is never computed, so it reports `0.0` as a sentinel and can trigger early stopping on the first epoch.

## Expected behaviour

- Setting `validation_fraction=0.0` must produce a 100% training / 0% validation partition with no validation pass.
- When `validation_fraction=0.0`, `early_stopping_metric` should automatically coerce to `TrainLoss`, or the combination must be rejected at submission time with a clear error.
- `FineTuneConfig::validate()` must reject the combination `validation_fraction=0.0` + `early_stopping_metric=ValLoss`.

## Suggested fix

Add a cross-field validation rule in `FineTuneConfig::validate()`:

    if self.validation_fraction == 0.0
        && self.early_stopping_metric == EarlyStoppingMetric::ValLoss
    {
        return Err(JammiError::FineTune(
            "early_stopping_metric=ValLoss requires validation_fraction > 0; \
             set early_stopping_metric=train_loss or set validation_fraction > 0".into(),
        ));
    }

Also ensure the Python client and gRPC converter always serialise `validation_fraction` explicitly (including when `0.0`) so the serde default is never silently applied.
